packages/@o3r/localization/src/devkit/localization-devtools.console.service.ts

```diff
@@ -1,21 +1,32 @@
 /* eslint-disable no-console */
-import { Inject, Injectable, Optional } from '@angular/core';
-import type { DevtoolsServiceInterface, WindowWithDevtools } from '@o3r/core';
-import { LocalizationDevtoolsServiceOptions } from './localization-devkit.interface';
+import { ApplicationRef, Inject, Injectable, Optional } from '@angular/core';
+import type { ContextualizationDataset, DevtoolsServiceInterface, WindowWithDevtools } from '@o3r/core';
+import { lastValueFrom, Subscription } from 'rxjs';
+import { take } from 'rxjs/operators';
+import { LocalizationService } from '../tools';
+import { LocalizationContextualizationDevtools, LocalizationDevtoolsServiceOptions } from './localization-devkit.interface';
 import { OtterLocalizationDevtools } from './localization-devtools.service';
 import { OTTER_LOCALIZATION_DEVTOOLS_DEFAULT_OPTIONS, OTTER_LOCALIZATION_DEVTOOLS_OPTIONS } from './localization-devtools.token';
 
 @Injectable()
-export class LocalizationDevtoolsConsoleService implements DevtoolsServiceInterface {
+export class LocalizationDevtoolsConsoleService implements DevtoolsServiceInterface, LocalizationContextualizationDevtools {
 
   /** Name of the Window property to access to the devtools */
   public static readonly windowModuleName = 'localization';
 
   constructor(
     private readonly localizationDevtools: OtterLocalizationDevtools,
+    private readonly localizationService: LocalizationService,
+    private readonly appRef: ApplicationRef,
     @Optional() @Inject(OTTER_LOCALIZATION_DEVTOOLS_OPTIONS) private options: LocalizationDevtoolsServiceOptions = OTTER_LOCALIZATION_DEVTOOLS_DEFAULT_OPTIONS
   ) {
-    if (this.options.isActivatedOnBootstrap) {
+    if (
+      this.options.isActivatedOnBootstrap
+      || (
+        this.options.isActivatedOnBootstrapWhenCMSContext
+        && (document.body.dataset as ContextualizationDataset).cmscontext === 'true'
```

Comment:
same here, is this possibly undefined ?

Reply:
https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/dataset
dataset is always present on every HTMLElement